demo/src/pick_place_task.cpp

```diff
@@ -0,0 +1,472 @@
+/*********************************************************************
+ * BSD 3-Clause License
+ *
+ * Copyright (c) 2019 PickNik LLC.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither the name of the copyright holder nor the names of its
+ *    contributors may be used to endorse or promote products derived from
+ *    this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Henning Kayser, Simon Goldstein
+   Desc:   A demo to show MoveIt Task Constructor in action
+*/
+
+#include <moveit_task_constructor_demo/pick_place_task.h>
+
+namespace moveit_task_constructor_demo
+{
+PickPlaceTask::PickPlaceTask(const std::string& task_name, const ros::NodeHandle& nh)
+  : nh_(nh), task_name_(task_name), execute_("execute_task_solution", true)
+{
+  ROS_INFO("waiting for task execution");
+  // execute_.waitForServer();
+
+  /****************************************************
+   *                                                  *
+   *               Load Parameters                    *
+   *                                                  *
+   ***************************************************/
+  ros::NodeHandle pnh("~");
+
+  // Planning group properties
+  group_name_ = pnh.param<std::string>("group_name", "manipulator");
```

Comment:
Let's remove all default values (at least setup-specific ones like "panda_hand" and "table_top") from here. People might try to copy and modify this and default parameters can make it hard to detect undesired behavior. Every task should have a well-defined set of properties so that we only need to change the config file to run it on another robot setup. I would even use [rosparam_shortcuts](https://github.com/PickNikRobotics/rosparam_shortcuts) to throw an error if something is missing.

Reply:
I updated those all